src/Eventso.Subscription/Observing/Batch/ErrorHandling/BreakdownFailedBufferProcessor.cs

```diff
@@ -0,0 +1,77 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Eventso.Subscription.Observing.Batch.ErrorHandling
+{
+    internal class BreakdownFailedBufferProcessor<TEvent> : IFailedBufferProcessor<TEvent>
+        where TEvent : IEvent
+    {
+        private readonly IEventHandler<TEvent> _handler;
+        private readonly IConsumer<TEvent> _consumer;
+
+        public BreakdownFailedBufferProcessor(IEventHandler<TEvent> handler, IConsumer<TEvent> consumer)
+        {
+            _handler = handler;
+            _consumer = consumer;
+        }
+
+        public async Task Process(
+            IReadOnlyList<Buffer<TEvent>.BufferedEvent> bufferedEvents,
+            CancellationToken cancellationToken)
+        {
+            SingleEventCollection convertibleCollection = null;
```

Comment:
if we got here then handling occured. that means that at least 1 not skipped event was in batch. so u will create SingleEventCollection anyway. so u can create empty collection, remove `is null` check on every iteration and use `SetEvent`

Reply:
I don't like it too, but I had to choose: complicated SingleEventCollection that supports zero size or complicated loop body. I tried both and decided that the second option is better)